circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -268,6 +271,29 @@ public void updateMetadata(
     }
   }
 
+  private Path createReplicaPartitionLocation(
+      TableAndStatistics sourceTableAndStatistics,
+      Partition sourcePartition,
+      ReplicaLocationManager locationManager) {
+    try {
+      return locationManager.getPartitionLocation(sourcePartition);
+    } catch (CircusTrainException e) {
```

Comment:
This is thrown when base location doesn't match partition location, strangely enough the actual data copy doesn't have a problem with this so this logic fixes it by generating the partition name (instead of taking the name from the source partition folder).

Reply:
Hmm I'm actually a bit concerned this might not work correctly with how this is implemented: https://github.com/HotelsDotCom/circus-train/blob/main/circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/source/HdfsSnapshotLocationManager.java#L124

I need to find out how the different copiers generate the target paths.